### PR TITLE
implement stock statistics

### PR DESCRIPTION
**Pull request by yatipatel**
Able to request data from api and store it in database (cache check is included).

### CODE REVIEW

**Comment by JScheire** [comment]
What did you use to test this?

**Comment by JScheire** [comment]
Can you just change the names of `models.js` and `routes.js` to like `equityModel.js` and `equityRoutes.js` or something like that

File: app/services/stock/routes/routes.js [hunk 1]
```diff
@@ -1 +1,12 @@
 const express = require('express');
+const router = express.Router();
+
+const equityController = require('../controllers/controller.js');
+
+router.get('', (req, res, next) => {
```
Comment (JScheire) [inline app/services/stock/routes/routes.js:6]: Is this just placeholder until we figure out a name for the endpoint?

**Comment by yatipatel** [comment]
> What did you use to test this?

We

File: app/services/stock/routes/routes.js (hunk 1 above)
Comment (christinem61) [reply app/services/stock/routes/routes.js:6]: yes

**Comment by yatipatel** [comment]
> What did you use to test this?

Not sure if this is what you mean, but we created a database and we were able to see what we stored and updated.

File: app/services/stock/controllers/controller.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                time: new Date() }
+            },
+            {
+                new: true,
+                upsert: true,
+            },
+        );
+        return savedEquity; 
+        } catch (err) {
+            console.log(err);
```
Comment (JScheire) [inline app/services/stock/controllers/controller.js:52]: can we add an appropriate error code with `res.status(CODE).send(err)`

File: app/services/stock/controllers/equityController.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+const fetch = require('node-fetch');
+const Equity = require("../models/equityModel.js");
+require('dotenv').config();
+
+const CACHE_TIME = 1;
+
+const getStatistics = async (ticker) => {
```
Comment (avidlaud) [inline app/services/stock/controllers/equityController.js:7]: I think this should be included in the try block

File: app/services/stock/index.js [hunk 4]
```diff
@@ -23,4 +21,6 @@ const connectDB = async () => {
 
 connectDB();
 
+app.use('/equity', stockRoute);
```
Comment (avidlaud) [inline app/services/stock/index.js:24]: I used the `/equity` endpoint for stock prices but we merge the two together into one routes file and just use both controllers.

File: app/services/stock/routes/equityRoutes.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const express = require('express');
+const router = express.Router();
+const equityController = require('../controllers/equityController.js');
+
+router.get('/:ticker', equityController.equity);
```
Comment (avidlaud) [inline app/services/stock/routes/equityRoutes.js:5]: Merge this file with stockRoutes.js in #38 to allow everything to use the `/equity` root. 

Currently have:
- `/equity/intraday/:ticker`
- `/equity/historical/:ticker`

Maybe use `/equity/statistics/:ticker` or something similar for `equityController.equity`?

File: app/services/stock/package-lock.json [hunk 6]
```diff
@@ -12,7 +12,9 @@
         "dotenv": "^8.2.0",
         "express": "^4.17.1",
         "jsonwebtoken": "^8.5.1",
-        "mongoose": "^5.11.15"
+        "mongoose": "^5.11.15",
+        "node-fetch": "^2.6.1",
+        "nodemon": "^2.0.7"
```
Comment (avidlaud) [inline app/services/stock/package-lock.json:17]: Not really important but not sure why this diff is so large - I guess nodemon adds a lot?

File: app/services/stock/controllers/equityController.js (hunk 3 above)
Comment (avidlaud) [reply app/services/stock/controllers/equityController.js:7]: ^The stuff below this

File: app/services/stock/controllers/controller.js (hunk 2 above)
Comment (christinem61) [reply app/services/stock/controllers/controller.js]: yes, done

File: app/services/stock/controllers/equityController.js (hunk 3 above)
Comment (christinem61) [reply app/services/stock/controllers/equityController.js:8]: yes, done

File: app/services/stock/routes/equityRoutes.js (hunk 5 above)
Comment (christinem61) [reply app/services/stock/routes/equityRoutes.js]: fixed

**Review by kjprajapati18 — APPROVED** [review]
I think just merge your index.js with what David has and we should be chilling. Otherwise, everything looks good!

**Review by avidlaud — DISMISSED** [review]

**Review by JScheire — DISMISSED** [review]

**Review by avidlaud — APPROVED** [review]

**Review by JScheire — APPROVED** [review]

**Review by kjprajapati18 — APPROVED** [review]

**Pull request merged**